### ISSUE TITLE
Question about the backproject function?

### ISSUE THREAD

**Issue by rockywind** [saic-vul/imvoxelnet#34]
Hello,
Why set the z>0 condition?
As I know, z is up,x is front, y is left in this coordinate. When set z>0 condition, it filters out all points below the ground.
But, sometimes the road is up and down.
  

> `def backproject(features, points, projection):
    n_images, n_channels, height, width = features.shape # [6, 64, 225, 400]
    n_x_voxels, n_y_voxels, n_z_voxels = points.shape[-3:]
    points = points.view(1, 3, -1).expand(n_images, 3, -1) #[6, 3, 1168128]
    points = torch.cat((points, torch.ones_like(points[:, :1])), dim=1) #[6, 4, 1168128]
    points_2d_3 = torch.bmm(projection, points) #[6, 3, 1168128]
    x = (points_2d_3[:, 0] / points_2d_3[:, 2]).round().long()
    y = (points_2d_3[:, 1] / points_2d_3[:, 2]).round().long()
    z = points_2d_3[:, 2]
    **valid = (x >= 0) & (y >= 0) & (x < width) & (y < height) & (z > 0)**
    volume = torch.zeros((n_images, n_channels, points.shape[-1]), device=features.device) #[6, 64, 312, 312, 12]
    for i in range(n_images):
        volume[i, :, valid[i]] = features[i, :, y[i, valid[i]], x[i, valid[i]]]
    volume = volume.view(n_images, n_channels, n_x_voxels, n_y_voxels, n_z_voxels)
    valid = valid.view(n_images, 1, n_x_voxels, n_y_voxels, n_z_voxels)
    return volume, valid`

**Comment by filaPro** [comment]
Hi @rockywind ,

After multiplying by projection matrix [here](https://github.com/saic-vul/imvoxelnet/blob/master/mmdet3d/models/detectors/imvoxelnet.py#L150) points are in camera coordinates. This means that `x` and `y` are now in image plane and `z` is the distance to the point. So [here](https://github.com/saic-vul/imvoxelnet/blob/master/mmdet3d/models/detectors/imvoxelnet.py#L154) with `z > 0` we simply check that point is in front of the camera (not behind).

**Comment by rockywind** [comment]
OK, I see.Thank you!
